lib/ClimaLandSimulations/src/utilities/pfts.jl

```diff
@@ -0,0 +1,614 @@
+"""
+This module is used to define plant functional types and how they can be used to
+build up the parameter sets for the model. A user can specify a percentage of
+land cover for each PFT within the domain over which they're running the model,
+and the parameters fed to the model are then the linear combination of the PFT 
+parameters weighted by the land cover percentages.
+"""
+
+export params_from_pfts, Pft, default_pfts
+
+using DataFrames
+using LinearAlgebra
+
+# List of parameters that PFTs need to define. Will become the column names in
+# the PFT DataFrame.
+param_list = [
+
+    # TwoStreamModel parameters
+    :Ω,          # Clumping Index
+    :α_PAR_leaf, # PAR Leaf Reflectance
+    :α_NIR_leaf, # Near-infrared Leaf Reflectance
+    :τ_PAR_leaf, # PAR Leaf Transmittance
+    :τ_NIR_leaf, # Near-infrared Leaf Transmittance
+
+    # Energy Balance model
+    :ac_canopy, # Specific Heat per Emmiting Area
+
+    # Conductance Model
+    :g1, # Medlyn CO2 Sensitivity
+
+    # Photosynthesis model
+    :Vcmax25, # Max rate of carboxylation of Rubisco
+
+    # Plant Hydraulics and general plant parameters
+    :SAI,             # Avg Stem Area Index
+    :f_root_to_shoot, # Root to Shoot Ratio
+    :K_sat_plant,     # Hydraulic Conductivity at Saturation
+    :ψ63,             # Xylem water potential at 63% loss of conductivity
+    :Weibull_param,   # Conductivity Weibull Parameter
+    :a,               # Retention Curve Curvature Parameter
+    :capacity,        # Water Storage Capacity
+    :plant_S_s,       # Specific Storativity
+    :rooting_depth    # Plant Rooting Depth
+]
+
+
+"""
+Define a PFT type that can be used to store the parameters for each PFT.
+Each PFT has a name and a list of parameters. The parameters are stored as 
+NamedTuples with the parameter name as the first element and the parameter value
+as the second element. The inner constructor checks to ensure all required
+parameters are defined for the PFT.
+"""
+struct Pft
+    name::String
+    parameters::NamedTuple
+
+    # Inner constructor to verify all required parameters are defined
+    function Pft(name, parameters)
+        defined_param_names = keys(parameters)
+        for param in param_list
+            if param ∉ defined_param_names
+                error("PFT $name is missing parameter $param")
+            end
+        end
+        new(name, parameters)
+    end
+end
+
+
+# The default PFTs used in the model use parameter values drawn from defferent 
```

Comment:
typo: different. 

Also I think it's okay to have this text in the docstring (what do you think @kmdeck)